### DISCUSSION TITLE
[Firebase_Messaging] - How can I disable the default notification on background message

### DISCUSSION THREAD

**Discussion by briankariuki** [firebase/flutterfire#5920] [Q&A]
I've setup a background message handler but I get two notifications on background message received. How can I disable the first notification and let the background handler show the notification?

```
static Future<dynamic> firebaseMessagingBackgroundHandler(RemoteMessage message) async {
    RemoteNotification notification = message.notification;
    final data = message.data;
    flutterLocalNotificationsPlugin.show(
      0,
      notification.title,
      notification.body,
      NotificationDetails(
        android: AndroidNotificationDetails(
          channel.id,
          channel.name,
          channel.description,
          icon: 'notification_icon',
          largeIcon: FilePathAndroidBitmap(
            await _downloadAndSaveFile(data["imageUrl"], 'largeIcon'),
          ),
          styleInformation: BigTextStyleInformation(
            notification.body,
            htmlFormatBigText: true,
            contentTitle: notification.title,
            htmlFormatContentTitle: true,
            summaryText: data["summary"],
            htmlFormatSummaryText: true,
          ),
          importance: Importance.max,
          priority: Priority.high,
        ),
      ),
    );
    print("onMessage: $message");

    return Future<void>.value();
  }
```

```
FirebaseMessaging.onBackgroundMessage(PushNotificationService.firebaseMessagingBackgroundHandler);
```

**Comment by QuentinSc** [comment]
Hello, remove the notification node from your payload

**Reply by Minituff** [reply]
This will work but removing that proprty will no longer guarantee delivery of the notification. See the [docs here](https://firebase.flutter.dev/docs/messaging/usage#low-priority-messages)

> As mentioned above, data only messages are classed as "low priority". Devices can throttle and ignore these messages if your application is in the background, terminated, or a variety of other conditions such as low battery or currently high CPU usage.
> 
> You should not rely on data only messages to be delivered, they should only be used to support your applications non-critical functionality, e.g. pre-fetching data so the next time the user opens your app the data is ready to be displayed and if the message never gets delivered then your app still functions and fetches data on open.
> 
> To help improve delivery, you can bump the priority of messages. Note; this does still not guarantee delivery.

**Comment by Minituff** [comment]
I actually just submitted an issue for this too: https://github.com/FirebaseExtended/flutterfire/discussions/6113
As far as I can tell. If you removed the `notification` property then you will get the answer you are looking for. However, removing the `notification` property will decrease the priority of the notification.

**Reply by briankariuki** [reply]
How can I increase the priority of the notification? Other apps use data only messages which get delivered all the time, can't this be achieved in flutter? I have data that needs to be shown on the notification, e.g an image

**Reply by Minituff** [reply]
When you send a notification to an FCM token there are a few properties that must be enabled. Refer to the docs [here](https://firebase.flutter.dev/docs/messaging/usage/#low-priority-messages):

```javascript
const admin = require("firebase-admin");

admin.initializeApp({
  credential: admin.credential.cert(require("./service-account-file.json")),
  databaseURL: "https://....firebaseio.com",
});

admin.messaging().send({
  token: "device token",
  data: {
    hello: "world",
  },
  // Set Android priority to "high"
  android: {
    priority: "high",
  },
  // Add APNS (Apple) config
  apns: {
    payload: {
      aps: {
        contentAvailable: true,
      },
    },
    headers: {
      "apns-push-type": "background",
      "apns-priority": "5", // Must be `5` when `contentAvailable` is set to true.
      "apns-topic": "io.flutter.plugins.firebase.messaging", // bundle identifier
    },
  },
});
```

`Note; this does still not guarantee delivery.`

**Comment by briankariuki** [comment]
A solution would be to create our own fcm message handler in the MainActivity of the app. This way would be able to get the payload in the data only field and create a notification with it. I'll look into this and share the code if succesful

**Reply by briankariuki** [reply]
Found a thread on stackoverflow with some insights on how one can implement this.

https://stackoverflow.com/questions/37711082/how-to-handle-notification-when-app-in-background-in-firebase

**Reply by QuentinSc** [reply]
Hello, I'm using fcm backgroundhandler and it seems android kill the bakground service after some days..
Don't find a solution that lasts over time

**Reply by QuentinSc** [reply]
https://github.com/FirebaseExtended/flutterfire/discussions/6042
